pennylane/qnode_new/jacobian.py

```diff
@@ -0,0 +1,528 @@
+                # the square of the observable
+                A = np.outer(A, A)
+                new = qml.expval(qml.ops.PolyXP(A, w, do_queue=False))
+
+            # replace the var(A) observable with <A^2>
+            self.circuit.update_node(e, new)
+            new_observables.append(new)
+
+        # calculate the analytic derivatives of the <A^2> observables
+        pdA2 = self._pd_parameter_shift(idx, args, kwargs, force_order2=(self.model == 'cv'))  # FIXME the force_order2 use here is convoluted and could be better
+
+        # restore the original observables, but convert their return types to expectation
+        for e, new in zip(var_observables, new_observables):
+            self.circuit.update_node(new, e)
+            e.return_type = ObservableReturnTypes.Expectation
+
+        # evaluate <A>
+        evA = np.asarray(self.evaluate(args, kwargs))
+
+        # evaluate the analytic derivative of <A>
+        pdA = self._pd_parameter_shift(idx, args, kwargs)
+
+        # restore return types
+        for e in var_observables:
+            e.return_type = ObservableReturnTypes.Variance
+
+        # return d(var(A))/dp = d<A^2>/dp -2 * <A> * d<A>/dp for the variances,
+        # d<A>/dp for plain expectations
+        return np.where(where_var, pdA2 -2 * evA * pdA, pdA)
+
+
+
+class AutogradQNode(JacobianQNode):
+    """JacobianQNode that works with Autograd."""
+
+    interface = 'Autograd'
+
+    # mark the evaluate method as an Autograd primitive
+    evaluate = autograd.extend.primitive(JacobianQNode.evaluate)
+
+    def __call__(self, *args, **kwargs):
+        args = autograd.builtins.tuple(args)  # prevents autograd boxed arguments from going through to evaluate
+        return self.evaluate(args, kwargs)
+
+    @staticmethod
+    def QNode_vjp(ans, self, args, kwargs):
+        """Returns the vector-Jacobian product operator for the QNode.
+
+        Takes the same arguments as :meth:`evaluate`, plus ``ans``.
+
+        Returns:
+            function[array[float], array[float]]: vector-Jacobian product operator
+        """
+        # pylint: disable=unused-argument
+        def gradient_product(g):
+            """Vector-Jacobian product operator.
+
+            Args:
+                g (array[float]): scalar or vector multiplying the Jacobian
+                    from the left (output side).
+
+            Returns:
+                nested Sequence[float]: vector-Jacobian product, arranged
+                into the nested structure of the input arguments in ``args``.
+            """
+            # Jacobian matrix of the circuit
+            jac = self.jacobian(args, kwargs)
+            if not g.shape:
+                temp = g * jac  # numpy treats 0d arrays as scalars, hence @ cannot be used
+            else:
+                temp = g @ jac
+
+            # restore the nested structure of the input args
+            temp = unflatten(temp.flat, args)
+            return temp
+
+        return gradient_product
```

Comment:
so much cleaner now!

Reply:
Yeah, it's neat to have all the Autograd stuff in one place, it gives you a much better idea of what's actually happening.